pkg/tiller/kind_sorter.go

```diff
@@ -123,5 +123,9 @@ func (k *kindSorter) Less(i, j int) bool {
 	if !ok {
 		return true
 	}
+	if first == second {
+		// same kind so sub sort alphanumeric
+		return a.name < b.name
+	}
```

Comment:
One more, for the next pass: When both kinds are unknown, sorting only by name still makes the output order dependent on the input order, because different objects of different kinds can have the same name. To cover that case, we could sort lexicographically first by kind name, and then by object name, or the converse. Right now, even with the change in f239050996f30901770314f68b425fcef29b19be, we'll wind up leaving same-named objects of different (unknown) kinds subject to input ordering.